lib/Frenetic_NetKAT_Virtual_Compiler.mli

```diff
@@ -1,15 +1,18 @@
 open Core.Std
 open Frenetic_NetKAT
+open Frenetic_NetKAT_FabricGen
 
-val compile : ?log:bool
-           -> ?record_paths:string
-           -> vrel:pred
-           -> vtopo:policy
-           -> ving_pol:policy
-           -> ving:pred
-           -> veg:pred
-           -> ptopo:policy
-           -> ping:pred
-           -> peg:pred
-           -> policy (* vpol, the virtual policy *)
-           -> policy
+module Make(FG:FABRIC_GEN) : sig
+  val compile : ?log:bool
+    -> ?record_paths:string
+    -> vrel:pred
+    -> vtopo:policy
+    -> ving_pol:policy
+    -> ving:pred
+    -> veg:pred
+    -> ptopo:policy
+    -> ping:pred
+    -> peg:pred
+    -> policy (* vpol, the virtual policy *)
+    -> policy
+end
```

Comment:
Very cool! 👍 Thanks for taking this on!
While we're at it, I'm wondering if we can extend this interface to three functions:
* generate_fabric (= FB.generate_fabric)
* compile_with_fabric
* compile (≈ generate_fabric |> compile_with_frabric)
The reason why this would be better is that fabric generation is typically costly, but it is **independent of the virtual policy**! So once can generate the fabric once and for all, and then use it with many different virtual policies.